src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -79,8 +79,6 @@ struct boss_baron_rivendare : public BossAI
 
     void JustDied(Unit* killer) override
     {
-        instance->SetData(TYPE_BARON, DONE);
```

Comment:
it is needed

Reply:
Isn't that the same here? Or is some different between `instance->SetData` and `instance->SetBossState`

```
void BossAI::_JustDied()
{
    events.Reset();
    summons.DespawnAll();
    scheduler.CancelAll();
    if (instance)
        instance->SetBossState(_bossId, DONE);
}
```

Reply:
it is not the same, stratholme uses the old and deprecated instance model

Reply:
Maybe something I should take a look. But for now I will change that back. Thanks for pointing